packages/compass-indexes/src/components/create-index-form/create-index-form.tsx

```diff
@@ -68,10 +85,33 @@ function CreateIndexForm({
         className={createIndexModalFieldsStyles}
         data-testid="create-index-form"
       >
-        <Body weight="medium" className={indexFieldsHeaderStyles}>
-          Index fields
-        </Body>
-        {fields.length > 0 ? (
+        {showIndexesGuidanceVariant ? (
+          <RadioBoxGroup
+            aria-labelledby="index-flows"
+            data-testid="create-index-form-flows"
+            id="create-index-form-flows"
+            onChange={(e) => {
+              onTabClick(e.target.value as Tab);
+            }}
+            value={currentTab}
+            className={createIndexModalFlowsStyles}
+          >
+            <RadioBox id="index-flow" value={'IndexFlow'}>
+              Start with an Index
+            </RadioBox>
+            <RadioBox id="query-flow" value={'QueryFlow'}>
+              Start with a Query
+            </RadioBox>
+          </RadioBoxGroup>
+        ) : (
+          <Body weight="medium" className={indexFieldsHeaderStyles}>
+            Index fields
+          </Body>
+        )}
+
+        {/* Only show the fields if user is in the Start with an index flow or if they're in the control */}
+        {fields.length > 0 &&
+        (showIndexesGuidanceVariant ? currentTab === 'IndexFlow' : true) ? (
```

Comment:
How do you feel about
```suggestion
        (!showIndexesGuidanceVariant || currentTab === 'IndexFlow') ? (
```

Reply:
makes sense, updated it!